src/backend/core/country/views.py

```diff
@@ -1,3 +1,81 @@
-# from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import permissions
+from django.conf import settings
 
-# Create your views here.
+from .models import (
+    Currency,
+)
+from .serializers import (
+    CurrencySerializer,
+)
+
+
+DEFAULT_LANGUAGE_CODE = settings.PARLER_DEFAULT_LANGUAGE_CODE
+LANGUAGES = settings.PARLER_LANGUAGES[None]
+# from parler.utils.conf import LanguagesSetting
+
+
+"""
+Language views
+"""
+
+
+class LanguagesView(APIView):
+    """
+    List all products for dashboard
+    """
+
+    # TODO: add permissions for dashboard views (only for staff) <- testing purposes
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        langs = [
+            {
+                "code": lang["code"],
+                "default": lang["code"] == DEFAULT_LANGUAGE_CODE,
+            }
+            for lang in LANGUAGES
+        ]
+        print(LANGUAGES)
+        return Response(langs, status=200)
+
+
+class CurrencyView(APIView):
+    """
+    List all products for dashboard
+    """
+
+    # TODO: add permissions for dashboard views (only for staff) <- testing purposes
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        qs = Currency.objects.all()
+        serializer = CurrencySerializer(qs, many=True)
+        return Response(serializer.data, status=200)
+
+    def post(self, request):
+        serializer = CurrencySerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=201)
+        return Response(serializer.errors, status=400)
+
+    def put(self, request, code):
+        try:
+            currency = Currency.objects.get(code=code)
+            serializer = CurrencySerializer(currency, data=request.data)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=200)
+            return Response(serializer.errors, status=400)
+        except Currency.DoesNotExist:
+            return Response(status=404)
+
+    def delete(self, request, code):
+        try:
+            currency = Currency.objects.get(code=code)
+            currency.delete()
```

Comment:
- I think we can use [Generic views](https://www.django-rest-framework.org/api-guide/generic-views/) for this, since the code is quite generic. And possibly we can override individual methods.
- maybe we should implement delete as "safe"

Reply:
> I think we can use [Generic views](https://www.django-rest-framework.org/api-guide/generic-views/) for this, since the code is quite generic. And possibly we can override individual methods.

Nice point, I'll do it

Reply:
> maybe we should implement delete as "safe"

Create please seperate issue from it. Let's make abstract `model` (and override `delete` method) from which we will derive all our `safe_delete` models

Reply:
> > I think we can use [Generic views](https://www.django-rest-framework.org/api-guide/generic-views/) for this, since the code is quite generic. And possibly we can override individual methods.
> 
> Nice point, I'll do it

Btw. this won't do much. We still have to provide post, get, delete and put methods.

Reply:
> > > I think we can use [Generic views](https://www.django-rest-framework.org/api-guide/generic-views/) for this, since the code is quite generic. And possibly we can override individual methods.
> > 
> > 
> > Nice point, I'll do it
> 
> Btw. this won't do much. We still have to provide post, get, delete and put methods.

Really? What's the problem there?
For example, look at [CategoryDetailViewDashboard](https://github.com/ecoseller/ecoseller/blob/master/src/backend/core/category/views.py#L47) that handles
- retrieve (get by ID)
- update
- delete
for me. I just needed to override method for deletion (to use safe delete)

Reply:
My fault, I was using `GenericAPIView` directly. This is really nice!

Reply:
Fixed, thanks ✔️